src/hb-ot-shape-complex-arabic-table.hh

```diff
@@ -139,28 +139,28 @@ joining_type (hb_codepoint_t u)
   switch (u >> 12)
   {
     case 0x0u:
-      if (hb_in_range (u, 0x0600u, 0x08E2u)) return joining_table[u - 0x0600u + joining_offset_0x0600u];
+      if (hb_in_range<hb_codepoint_t> (u, 0x0600u, 0x08E2u)) return joining_table[u - 0x0600u + joining_offset_0x0600u];
```

Comment:
I was doing this but found out U already has a conflicting [use](https://github.com/behdad/harfbuzz/blob/02bfd965af588640250c47f344ad3096c8a373e0/src/hb-ot-shape-complex-arabic-table.hh#L23).

Not in favor of u() personally as it will obfuscating things more on my view, lets defer it if is possible.